clarity/contracts/arkadiko-stacker-v2-1.clar

```diff
@@ -0,0 +1,248 @@
+          result (begin
+            (print result)
+            (print (stx-account (as-contract tx-sender)))
+            (var-set previous-stacking-unlock-burn-height (get unlock-burn-height result))
+            (var-set stacking-unlock-burn-height (get unlock-burn-height result))
+            (var-set stacking-stx-stacked (get lock-amount result))
+            (try! (contract-call? .arkadiko-freddie-v1-1 set-stacking-unlock-burn-height (var-get stacker-name) (get unlock-burn-height result)))
+            (asserts! (is-ok (update-stacker-variables)) (err u0))
+            (ok (get lock-amount result))
+          )
+          error (begin
+            (print (err (to-uint error)))
+          )
+        )
+      )
+      failure (print (err (to-uint failure)))
+    )
+  )
+)
+
+(define-private (update-stacker-variables)
+  (let (
+    (name (var-get stacker-name))
+  )
+    (if (is-eq name "stacker")
+      (begin
+        (var-set stacker-name "stacker-2")
+        (try! (contract-call? .arkadiko-stx-reserve-v1-1 set-next-stacker-name "stacker-2"))
+      )
+      (if (is-eq name "stacker-2")
+        (begin
+          (var-set stacker-name "stacker-3")
+          (try! (contract-call? .arkadiko-stx-reserve-v1-1 set-next-stacker-name "stacker-3"))
+        )
+        (if (is-eq name "stacker-3")
+          (begin
+            (var-set stacker-name "stacker-4")
+            (try! (contract-call? .arkadiko-stx-reserve-v1-1 set-next-stacker-name "stacker-4"))
+          )
+          (if (is-eq name "stacker-4")
+            (begin
+              (var-set stacker-name "stacker")
+              (try! (contract-call? .arkadiko-stx-reserve-v1-1 set-next-stacker-name "stacker"))
+            )
+            true
+          )
+        )
+      )
+    )
+    (ok true)
+  )
+)
+
+(define-read-only (total-tokens-to-stack)
+  (let (
+    (tokens-to-stack (unwrap! (contract-call? .arkadiko-stx-reserve-v1-1 get-tokens-to-stack "stacker") (ok u0)))
+    (tokens-to-stack-2 (unwrap! (contract-call? .arkadiko-stx-reserve-v1-1 get-tokens-to-stack "stacker-2") (ok u0)))
+    (tokens-to-stack-3 (unwrap! (contract-call? .arkadiko-stx-reserve-v1-1 get-tokens-to-stack "stacker-3") (ok u0)))
+    (tokens-to-stack-4 (unwrap! (contract-call? .arkadiko-stx-reserve-v1-1 get-tokens-to-stack "stacker-4") (ok u0)))
+  )
+    (ok (+ tokens-to-stack tokens-to-stack-2 tokens-to-stack-3 tokens-to-stack-4))
+  )
+)
+
+;; should be called to add additional STX tokens stacking
+;; if there is net outflow of STX between two cycles, does not need to be called
+;; before calling this, consolidate the new amount of tokens to stack in PoX in stx-reserve `set-tokens-to-stack`
+;; then call this first, before a new cycle starts (every 2100 blocks)
+;; after calling this, call `stack-extend`
+(define-public (stack-increase (for-stacker (string-ascii 256)) (additional-tokens-to-stack uint))
```

Comment:
Q: How do you calculate 'additional-tokens-to-stack'?  Using `get-tokens-to-stack` and what is already stacking for a given stacker? Can it be done in this method instead of off-chain? 
I'm just wondering if it's possible to use the Keepers to automate stacking to avoid us managing this manually.

Reply:
@nieldeckx Probably can be done off-chain.. but the more we do in contracts, the more things can go wrong and since we have to initiate a call every 2 weeks anyway, I thought of either guesstimating it or just writing an off-chain script :) but if it can be done on-chain, feel free to give it a try!

Reply:
I think `(additional-tokens-to-stack (- (total-tokens-to-stack) (var-get stacking-stx-stacked)))`. But not confident enough to just change this with limited tests..